docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-
-!!!
-
-!!! note
-
-Custom TA
-
-The simplest way to add a custom TA is to derive it from the existing implementation of the TCP or BT adapter and add it to the TM
-
-!!!
-
-```
-#include "transport_manager.h"
-#include "transport_manager_default.h"
-
-{
-    // The default implementation of TCP adapter is used in this example
-    // The developer is free to create their own implementation
-    TransportAdapter *my_transport_adapter = static_cast<TransportAdapter*>(new TcpTransportAdapter);
-    my_transport_adapter->init(); // note: TA must be initialized before getting an instance of TM
-    TransportManager *tm_impl = TransportManagerDefault::instance(); // note: When custom TA is added to the TM, the default listener will be assigned to it automatically
-    tm_impl->addTransportAdapter(my_transport_adapter)
-}
-
-```
+#### Transport Adapter Structure
 
-### Initialization
+|||
+Figure 3: Transport Adapter UML Diagram
+![TA](./assets/ta.png)
+|||
 
-By default, the TA is initialized during TM initialization. If the TA needs to be added to a TM which has already been initialized, the TA should be initialized before it is added the TM
+### Transport Adapter Workers
 
-!!! note
+The Device Scanner is responsible for scanning for new devices to connect with. When a device is found, this worker is responsible for alerting the Transport Adapter, as well as alerting the Transport Manager via the Transport Adapter Listener. Next, the Transport Manager will instruct the Transport Adapter to connect with the devices.
 
-The TM has a TCP adapter by default. Adding the new instance of the same Adapter may result in unexpected behavior. The developer must change the standard behavior of the TCP adapter to elinatate any potential problems.
+The Client Connection Listener implements receiving a connection that is originated by a device. This will typically wait for connection from a device, then establish that connection, finally alerting the Transport Manager via the Transport Manager Listener of the newly connected device and app IDs.
 
-!!!
+The Server Connection Factory implements a connection that is originated from Core. For example, Core reaches out to a predefined web address to start a cloud application. This type of communication requires that the Transport Adapter knows of the device and application in advance. When this connection is created, the Transport Adapter will alert the Transport Manager of the new devices and applications in a similar fashion to other workers.
 
-### Custom Transport Adapter Implementations
+Depending on what your type of transport is, whether Core will be the server or the client, you will likely implement either the Client Connection Listener or the Server Connection Factory.
 
-If the default implementation of TAs is not suitable, you can create a transport adapter from scratch.
+#### Client Connection Listener
```

Comment:
I would put these TA worker sections in the order that you first mentioned them above.